crawl-ref/source/l-travel.cc

```diff
@@ -8,10 +8,48 @@
 #include "branch.h"
 #include "cluautil.h"
 #include "coord.h"
+#include "env.h"
 #include "player.h"
 #include "terrain.h"
 #include "travel.h"
 
+static bool _in_map_bounds(int x, int y)
+{
+    // code to get bounds stolen from viewmap.cc
```

Comment:
Please extract this into a function shared by both uses instead of copying it. I'd suggest that it return a pair of coordinates forming a bounding box, as the existing code in `show_map` needs; then your usage here can just be a small wrapper around that that checks whether your target point is inside the box. It's slightly more expensive than what you're written now, since you won't be able to short-circuit, but that's not as important here as avoiding duplication.

Reply:
was going to do that originally but got bummed out because returning multiple values from a function sucks before c++17, will change

Reply:
Changed